src/hardware/siglent-sdl10x0/api.c

```diff
@@ -0,0 +1,461 @@
+	SR_CONF_CURRENT | SR_CONF_GET,
+	SR_CONF_CURRENT_LIMIT | SR_CONF_GET | SR_CONF_SET | SR_CONF_LIST,
+	SR_CONF_POWER | SR_CONF_GET,
+	SR_CONF_POWER_TARGET | SR_CONF_GET | SR_CONF_SET | SR_CONF_LIST,
+	SR_CONF_RESISTANCE | SR_CONF_GET,
+	SR_CONF_RESISTANCE_TARGET | SR_CONF_GET | SR_CONF_SET | SR_CONF_LIST,
+	SR_CONF_OVER_POWER_PROTECTION_ENABLED | SR_CONF_GET,
+	SR_CONF_OVER_POWER_PROTECTION_ACTIVE | SR_CONF_GET,
+	SR_CONF_OVER_POWER_PROTECTION_THRESHOLD | SR_CONF_GET | SR_CONF_SET,
+	SR_CONF_OVER_CURRENT_PROTECTION_ENABLED | SR_CONF_GET,
+	SR_CONF_OVER_CURRENT_PROTECTION_ACTIVE | SR_CONF_GET,
+	SR_CONF_OVER_CURRENT_PROTECTION_THRESHOLD | SR_CONF_GET | SR_CONF_SET,
+};
+
+static const char *regulation[] = {
+	"CURRENT",
+	"VOLTAGE",
+	"POWER",
+	"RESISTANCE",
+	"LED"
+};
+
+static struct sr_dev_driver siglent_sdl10x0_driver_info;
+
+static struct sr_dev_inst *probe_device(struct sr_scpi_dev_inst *scpi)
+{
+	struct sr_dev_inst *sdi;
+	struct sr_channel_group *cg;
+	struct sr_channel *ch;
+	struct dev_context *devc;
+	struct sr_scpi_hw_info *hw_info;
+
+	sdi = NULL;
+	devc = NULL;
+	hw_info = NULL;
+
+	if (sr_scpi_get_hw_id(scpi, &hw_info) != SR_OK) {
+		sr_info("Couldn't get IDN response.");
+		goto fail;
+	}
+
+	if (std_str_idx_s(hw_info->manufacturer, ARRAY_AND_SIZE(manufacturers)) < 0)
+		goto fail;
+
+	if (std_str_idx_s(hw_info->model, ARRAY_AND_SIZE(models)) < 0)
+		goto fail;
+
+	sdi = g_malloc0(sizeof(struct sr_dev_inst));
+	sdi->vendor = g_strdup(hw_info->manufacturer);
+	sdi->model = g_strdup(hw_info->model);
+	sdi->version = g_strdup(hw_info->firmware_version);
+	sdi->serial_num = g_strdup(hw_info->serial_number);
+	sdi->driver = &siglent_sdl10x0_driver_info;
+	sdi->inst_type = SR_INST_SCPI;
+	sdi->conn = scpi;
+	sdi->channels = NULL;
+	sdi->channel_groups = NULL;
+
+	cg = sr_channel_group_new(sdi, "1", NULL);
+
+	ch = sr_channel_new(sdi, 0, SR_CHANNEL_ANALOG, TRUE, "1");
+	cg->channels = g_slist_append(cg->channels, ch);
+
+	sr_scpi_hw_info_free(hw_info);
+	hw_info = NULL;
+
+	devc = g_malloc0(sizeof(struct dev_context));
+	sdi->priv = devc;
+
+	// modelname 'SDL1020X-E': 6th character indicates wattage: 2 => 200, 3 => 300
+	devc->maxpower = 200.0;
+	if (g_ascii_strncasecmp(sdi->model, "SDL1030", strlen("SDL1030")) == 0) {
+		devc->maxpower = 300.0;
+	}
+
+
+	return sdi;
+
+fail:
+	sr_scpi_hw_info_free(hw_info);
+	sr_dev_inst_free(sdi);
+	g_free(devc);
+
+	return NULL;
+}
+
+static GSList *scan(struct sr_dev_driver *di, GSList *options)
+{
+	return sr_scpi_scan(di->context, options, probe_device);
+}
+
+static int dev_open(struct sr_dev_inst *sdi)
+{
+	return sr_scpi_open(sdi->conn);
+}
+
+static int dev_close(struct sr_dev_inst *sdi)
+{
+	return sr_scpi_close(sdi->conn);
+}
+
+static int config_get(uint32_t key, GVariant **data,
+	const struct sr_dev_inst *sdi, const struct sr_channel_group *cg)
+{
+	int ret;
+	float ival;
+	gboolean bval;
+	char *mode;
+
+	(void)sdi;
+	(void)data;
+	(void)cg;
```

Comment:
These (void) statements are just in the stubs generated by `new-driver` to make the generated patch compile without warnings about unused variable. You're supposed to remove them as you fill out the stubs (other functions too).